### DISCUSSION TITLE
Hubspot API Integration question

### DISCUSSION THREAD

**Discussion by nickfmc** [solspace/craft-freeform#261] [Questions & Answers] [answered]
Would the Hubspot API integration allow you to create a form with contact info and 3 checkboxes for mailing lists and add/update the status of a users list activity based on form submission? so if a user was in list 1 but they resubmitted the form and checked of list 2 and list 3 can we remove them from list one and add them to list 2 and 3 on submission?

Email
- [ ] List 1
- [ ] List 2
- [ ] List 3

**Comment by kjmartens** [comment, marked as answer]
Hi @nickfmc and sorry for the delay...

There is a duplicate check and update available to the HubSpot integration, and it works like this:
https://docs.solspace.com/craft/freeform/v3/integrations/crm/hubspot.html#duplicate-check-update

If it detects a duplicate, it will either replace the options entirely or just append additions this time around, depending on whether or not you have the **Append checkbox group field values on Contact/Company update?** settings enabled in Freeform. So the answer to your question is that you have the choice of how you wish for this to behave. :)

**Reply by nickfmc** [reply]
great thanks!
